project.mml

```diff
@@ -1035,11 +1035,41 @@ Layer:
         (SELECT
             way,
             aeroway,
-            bridge IN ('yes', 'boardwalk', 'cantilever', 'covered', 'low_water_crossing', 'movable', 'trestle', 'viaduct') AS bridge
-          FROM planet_osm_line
-          WHERE aeroway IN ('runway', 'taxiway')
-          ORDER BY bridge NULLS FIRST,
-            CASE WHEN aeroway = 'runway' THEN 1 ELSE 0 END
+            runway,
+            bridge,
+            width,
+            width::real/sf AS area_width,
+            1/sf AS line_width
+          FROM
+          (SELECT
+              way,
+              aeroway,
+              tags->'runway' AS runway,
+              bridge IN ('yes', 'boardwalk', 'cantilever', 'covered', 'low_water_crossing', 'movable', 'trestle', 'viaduct') AS bridge,
+              -- Convert valid length units into metres. Fall back to a sensible default if no valid value was set.
+              GREATEST(
+                CASE
+                    -- Metric:
+                    WHEN tags->'width' ~ '^[0-9]*[.]?[0-9]+ m$' THEN LEFT(tags->'width', -2)::real
+                    WHEN tags->'width' ~ '^[0-9]*[.]?[0-9]+ km$' THEN LEFT(tags->'width',-3)::real * 1000
+                    WHEN tags->'width' ~ '^[0-9]*[.]?[0-9]+$' THEN (tags->'width')::real
+                    -- Miles and nautical miles:
+                    WHEN tags->'width' ~ '^[0-9]*[.]?[0-9]+ mi$' THEN LEFT(tags->'width', -3)::real * 1609.344
+                    WHEN tags->'width' ~ '^[0-9]*[.]?[0-9]+ nmi$' THEN LEFT(tags->'width', -4)::real * 1852
+                    -- Feet and inches:
+                    WHEN tags->'width' ~ '^[0-9]+''[0-9]+"$' THEN (SPLIT_PART(tags->'width', '''', 1)::real * 12 + LEFT(SPLIT_PART(tags->'width', '''', 2), -1)::real) * 0.0254
+                    WHEN tags->'width' ~ '^[0-9]+''$' THEN LEFT(tags->'width', -1)::real * 12 * 0.0254
+                    WHEN tags->'width' ~ '^[0-9]+"$' THEN LEFT(tags->'width', -1)::real * 0.0254
+                    ELSE NULL
```

Comment:
This was tested with this input:

### Valid:

```
50
50 m
0.05 km
.05 km
0.03 mi
0.03 nmi
164'1"
50.0
164'
1969"
```

### Broken (returns `NULL`)
 
```
AA"
AA
AA.A
.A
.A m
A m
A.A m
```